action/evm_transaction.go

```diff
@@ -22,17 +23,32 @@ type (
 		inner TxData
 	}
 
+	// TxData is the interface required to execute a transaction by EVM
+	// It follows the same-name interface in go-ethereum
 	TxData interface {
-		Nonce() uint64
-		GasLimit() uint64
-		GasPrice() *big.Int
-		Amount() *big.Int
+		TxCommon
+		Value() *big.Int
 		To() *common.Address
 		Data() []byte
+	}
+
+	TxCommon interface {
+		Nonce() uint64
+		Gas() uint64
+		GasPrice() *big.Int
```

Comment:
TODO: dynamic fee related methods and blob related methods will be added into `TxCommon`